### ISSUE TITLE
Dynabosses health increases exponentially during fight

### ISSUE THREAD

**Issue by reneraen** [project-rubika/issues#477]
**Describe the bug**
When approaching a dyna, during the fight unless you kill it quite quickly the health is recalculated to a number that is not possible to kill for normal players. I've attached coordinates (shift+f9) at bottom to one of the dynacamp locations in the world, but any should suffice. We even reproduced this in the backyards.

**To Reproduce**
1. Go to any place where a dynaboss is (backyard is fine)
2. Engage the mob, let the mob attack you but dont attack back.
3. Do /get health
4. See that the stats for health increases exponentially to such an extent that the boss becomes unkillable

**Expected behavior**
That the boss HP doesnt recalculate and increase during fight.

**Screenshots**
<img width="907" alt="image" src="https://github.com/project-rubika/issues/assets/48966634/74757482-68fb-40dc-a722-ceea98bdf6cf">

**Character info:**

```
(22:21) Madcow [300 Keeper]
------------------------------------
    strength 	2626
    stamina 	2634
    agility 	2466
    sense 	2454
    psychic 	2506
    intelligence 	2465
    flags 	528961
    monsterscale 	100
    currentnano 	405793
    headmesh 	40704
    cash 	900227968
    gmlevel 	127
    expansion 	3
    breed 	4
    visualbreed 	4
    profession 	14
    visualprofession 	14
    level 	300
    sex 	1
    visualsex 	1
    titlelevel 	7
    health 	2234283
    equippedweapontype 	130
    EquippedRHWeapon 	130
    selectedtarget 	772
    currentncu 	1
    currentmovementmode 	3
    waitstate 	2
    selectedtargettype 	0
    isfightingme 	2
    charstate 	0
    xp 	75920900
    nextxp 	75920900
    lastsavexp 	6500
    lastxp 	75920900
    computerliteracy 	4912
    lastsaved 	75920900
    exitstatel 	-1073675336
    externalplayfieldinstance 	0
    externaldoorinstance 	0
    playfieldtype 	0
    treatment 	4890
    martialarts 	6871
    meleemultiple 	7431
    onehandbluntweapons 	8028
    onehandedgedweapon 	7440
    piercing 	7436
    twohandbluntweapons 	7686
    twohandedgedweapons 	7456
    meleeenergyweapon 	6881
    closecombatinitiative 	6615
    physicalprowessinitiative 	6115
    bodydevelopment 	5146
    nanoenergypool 	4371
    nanoac 	11256
    parry 	7434
    duck 	6914
    dodge 	6914
    evade 	6914
    psychology 	4714
    brawl 	5132
    specialization 	15
    shadowbreedtemplate 	230908
    alienlevel 	30
    riposte 	4940
    dimach 	4891
    adventuring 	4857
    swim 	4869
    sneakattack 	4890
    fastattack 	4890
    throwingknife 	4898
    grenade 	4615
    throwngrapplingweapons 	4824
    bow 	4843
    pistol 	4620
    assaultrifle 	4641
    submachinegun 	4645
    shotgun 	4637
    rifle 	4675
    lr_energyweapon 	4679
    flingshot 	4616
    aimedshot 	5050
    burst 	4636
    fullauto 	4657
    bowspecialattack 	4894
    lr_multipleweapon 	4616
    distanceweaponinitiative 	3816
    nanoprowessinitiative 	5014
    mechanicalengineering 	4753
    electricalengineering 	4636
    fieldquantumphysics 	4621
    weaponsmithing 	4911
    pharmaceuticals 	4616
    nanoprogramming 	4616
    chemistry 	4637
    tutoring 	4836
    materialmetamorphose 	4893
    biologicalmetamorphose 	4905
    psychologicalmodification 	4890
    materialcreation 	4911
    materiallocation 	4903
    senseimprovement 	4899
    firstaid 	4835
    concealment 	4697
    breakingentry 	4618
    disarmtrap 	4614
    perception 	4889
    driveair 	4614
    driveground 	4752
    drivewater 	4614
    mapnavigation 	4698
    maxnanoenergy 	2009890
    fatness 	1
    alignment 	0
    side 	0
    maxncu 	4668
    race 	1
    accountflags 	1234567890
    life 	2135684
    monsterdata 	0
    VisualFlags 	31
    runspeed 	3427
    clan 	0
    projectileac 	47257
    energyac 	47257
    meleeac 	47257
    coldac 	47257
    fireac 	47257
    chemicalac 	47407
    radiationac 	47257
    poisonac 	47407
    healdelta 	100202
    shieldprojectileac 	123
    shieldmeleeac 	123
    shieldenergyac 	123
    shieldchemicalac 	123
    shieldradiationac 	123
    shieldcoldac 	123
    shieldnanoac 	80
    shieldfireac 	123
    shieldpoisonac 	123
    reflectmeleeac 	1003
    reflectpoisonac 	1003
    reflectfireac 	1003
    reflectnanoac 	1003
    reflectcoldac 	1003
    reflectprojectileac 	1003
    reflectenergyac 	1003
    reflectchemicalac 	1003
    reflectradiationac 	1003
    healinterval 	1
    nanointerval 	1
    meleedamagemodifier 	87
    poisondamagemodifier 	69
    firedamagemodifier 	87
    colddamagemodifier 	50
    projectiledamagemodifier 	50
    energydamagemodifier 	50
    chemicaldamagemodifier 	69
    radiationdamagemodifier 	61
    team 	0
    beltslots 	6
    npcostmodifier 	-2
    criticalincrease 	15
    rangeincreaserweapon 	45
    amsmodifier 	101
    dmsmodifier 	343
    xpmodifier 	3
    aggdef 	100
    state 	0
    unarmedtemplateinstance 	0
    invaderskilled 	0
    killedbyinvaders 	0
    vp 	0
    unsavedxp 	0
    nanofocuslevel 	0
    shadowbreed 	33
    lastperkresettime 	0
    socialstatus 	0
    playeroptions 	0
    tempsaveteamid 	0
    tempsaveplayfield 	0
    tempsavex 	0
    tempsavey 	0
    PVPDuelKills 	0
    PVPDuelDeaths 	0
    PVPProfessionDuelKills 	0
    PVPProfessionDuelDeaths 	0
    PVPRankedSoloKills 	0
    PVPRankedSoloDeaths 	0
    PVPRankedTeamKills 	0
    PVPRankedTeamDeaths 	0
    PVPSoloScore 	0
    PVPTeamScore 	0
    PVPDuelScore 	0
    unreadmailcount 	0
    lastmailchecktime 	1283065897
    savedxp 	0
    features 	6
    apartmentsallowed 	1
    apartmentsowned 	0
    lastconcreteplayfieldinstance 	0
    mapoptions 	0
    mapareapart1 	0
    mapareapart2 	0
    mapareapart3 	0
    mapareapart4 	0
    missionbits1 	0
    missionbits2 	0
    missionbits3 	0
    missionbits4 	0
    missionbits5 	0
    missionbits6 	0
    missionbits7 	0
    missionbits8 	0
    missionbits9 	0
    missionbits10 	0
    missionbits11 	0
    missionbits12 	0
    sessiontime 	1234567890
    autoattackflags 	5
    personalresearchlevel 	0
    globalresearchlevel 	0
    personalresearchgoal 	0
    globalresearchgoal 	0
    battlestationside 	0
    battlestationrep 	10
    members 	999
    veteranpoints 	0
    monthspaid 	0
    PaidPoints 	0
    xpkillrange 	5
    inplay 	0
    attitude 	0
    icon 	0
    mesh 	17530
    deadtimer 	0
    currentmass 	0
    itemtype 	0
    previoushealth 	50
    currentstate 	0
    age 	0
    attackspeed 	5
    metatype 	0
    rp 	0
    specialcondition 	1
    sk 	0
    clanredeemed 	0
    clanconserver 	0
    clandevoted 	0
    otunredeemed 	0
    otoperator 	0
    otfollowers 	0
    gos 	0
    clanvanguards 	0
    ottrans 	0
    clangaia 	0
    otmed 	1234567890
    clansentinels 	0
    otarmedforces 	0
    playerid 	1234567890
    aliennextxp 	1500
    alienxp 	0
    insurancepercentage 	0
    prevmovementmode 	3
    teamside 	0
    absorbprojectileac 	0
    absorbmeleeac 	0
    absorbenergyac 	0
    absorbchemicalac 	0
    absorbradiationac 	0
    absorbcoldac 	0
    absorbnanoac 	0
    absorbfireac 	0
    absorbpoisonac 	0
    temporaryskillreduction 	0
    insurancetime 	0
    mapflags 	0
    changesidecount 	0
    reflectreturnedmeleeac 	0
------------------------------------

(22:21) .getfull succeeded.

```

**Playfield info:**

```
N/A - It applies to any zone where a dynaboss is

(22:16) Extended location information:
(22:16) - 3498.1, 1891.9, 40.0 (3498.1 1891.9 y 40.0 590)
(22:16) - Pf Proxy: Model=51100:590 GS=0 SG=0 R=590, resource: 590
(22:16) - zone: 4599, area: " Eastern Swampland"
(22:16) Server id: 772, character id: 50000:772, time: 2024-01-16 21:16:03 (UTC)
(22:16) Version: 00.4.4_EP1, map id: NOT FOUND, build: 52606.


```
----

**Comment by TheTinkeringIdiot** [comment]
Confirmed.

It appears as if the boss is in combat (actively attacking) but regaining health as if it is not, and the health it regains is added to its maximum health. I witnessed small increases (normal for a healdelta of 1 which this test boss had) intermixed with large jumps resembling out-of-combat NPC healing. Values reported by `/get health` continue to climb indefinitely even though this boss was not damaged at all and has no health deficit to regain. 

Testing with a normal mob indicates that this effect is limited to boss types. 

![image](https://github.com/project-rubika/issues/assets/96996793/959ace32-dd11-4b2b-9c6c-3093f446248f)
